system/modules/core/classes/Backend.php

```diff
@@ -801,17 +795,20 @@ public static function addFileMetaInformationToRequest($strUuid, $strPtable, $in
 				break;
 		}
 
-		if ($objPage === null || $objPage->numRows < 1)
+		if ($objPage === null || ($objPage instanceof \Database\Result && $objPage->numRows < 1))
 		{
 			return;
 		}
 
-		$objModel = new \PageModel();
-		$objModel->setRow($objPage->row());
-		$objModel->loadDetails();
+		if (!$objPage instanceof \PageModel)
+		{
+			$objPage = \PageModel::findByPk($objPage->id);
```

Comment:
I have already merged the PR, but now I noticed that we are not checking `!== null` here anymore. Can I change it to the following?

```php
	default:
		// HOOK: support custom modules
		if (isset($GLOBALS['TL_HOOKS']['addFileMetaInformationToRequest']) && is_array($GLOBALS['TL_HOOKS']['addFileMetaInformationToRequest']))
		{
			foreach ($GLOBALS['TL_HOOKS']['addFileMetaInformationToRequest'] as $callback)
			{
				if (($val = \System::importStatic($callback[0])->{$callback[1]}($strPtable, $intPid)) !== false)
				{
					$objPage = $val;
				}
			}
					
			if ($objPage instanceof \Database\Result && $objPage->numRows < 1)
			{
				return;
			}

			if (!$objPage instanceof \PageModel)
			{
				$objPage = \PageModel::findByPk($objPage->id);
			}
		}
		break;
}

if ($objPage === null)
{
	return;
}

$objPage->loadDetails();
```